.github/workflows/pre-commit.yaml

```
@@ -25,19 +25,32 @@ jobs:
     - name: Install shellcheck
       run: |
         sudo apt update && sudo apt install shellcheck
+
+    - name: Install hadolint
+      run: |
+        curl -L "$(curl -s https://api.github.com/repos/hadolint/hadolint/releases/latest | grep -o -E -m 1 "https://.+?/hadolint-Linux-x86_64")" > hadolint \
+        && chmod +x hadolint && sudo mv hadolint /usr/bin/
```

Not essential but makes use of single command
```suggestion
        && sudo install -m 0755 -o root -g root hadolint /usr/bin/
```

---

Hmm, this is the first time I've seen such an approach.
If the command took up less space, you could think about using it

---

> Hmm, this is the first time I've seen such an approach.

There's always a space to learn. Just like the `mapfile` command 😏
`install` is more commonly used within packages' installation scripts and (just like `mapfile`) isn't commonly known by regular users.

> If the command took up less space, you could think about using it

This is not about space consumption, but about proper file installation including mode, user, and group (it is not a good practice to have user-owned files in system locations) along with using a single command which is designed specifically for the use case we have here instead if using chmod/chown/mv.

Again: this is not essential. Up to you to consider.

---

> If the command took up less space, you could think about using it

btw, the `mapfile` change is 9 chars longer than what existed before 😏